src/icrc1_auction_api.mo

```diff
@@ -693,6 +697,10 @@ actor class Icrc1AuctionAPI(quoteLedger_ : ?Principal, adminPrincipal_ : ?Princi
 
   private func runAuction() : async () {
     let a = U.requireMsg(auction, "Not initialized");
+    if (nextAssetIdToProcess == 0) {
+      let startTimeDiff : Int = Nat64.toNat(Prim.time() / 1_000_000) - nextSessionTimestamp() * 1_000;
```

Comment:
This does not handle a negative diff. It will trap.

Reply:
Seems to work for me https://embed.motoko.org/motoko/g/BeKuhd8UQxqibAG2n2zBANuHTMpPSdHtd1NBGZgQCPE1xhMq9c42grptckHjbT7ttXXH4vc4EsphaUMLjQd7uSRLk?lines=5

Reply:
Ah sorry, misread the expression.